src/messaging/BUILD.gn

```diff
@@ -77,4 +77,8 @@ static_library("messaging") {
     "${chip_root}/src/transport",
     "${chip_root}/src/transport/raw",
   ]
+
+  if (chip_enable_icd_server) {
+    public_deps += [ "${chip_root}/src/app/icd:data" ]
```

Comment:
should icd:notifier also be guarded by chip_enable_icd_server ?

Reply:
At some point, yes. Right now, the icd:notifier source_set has the dependecy to the app_config which contains the define that gates all the icd calls, `CHIP_CONFIG_ENABLE_ICD_SERVER`. If we remove the icd:notifier, non-icd builds fail because of the undef error.

We still need to find a better location for the configuration define so leaving it as is for now.
https://github.com/project-chip/connectedhomeip/issues/30630

Reply:
The two viable options i see are :

1. Have an `config("icd_config") that can be added as a `public_deps` to anyone that needs to be aware of define
2. create an icd_buildconfig (like app_buildconfig) and add that as a dependency to anyone that needs the define

Not sure which is best...

Reply:
PR to fix this #30661